### PR TITLE
Migrate team to hexathons

### PR DESCRIPTION
**Pull request by Roshan-D**
### Description
Moved `/team` routes and `Team` modal from the `users` service to the `hexathons` service

### Issues
Addresses #179 
-

### CODE REVIEW

**Review by amtang325 — COMMENTED** [review]
looks good!

**Review by ayush-goyal — CHANGES_REQUESTED** [review]
A few things, this is gonna break the registration team flow so make sure that’s fixed before you merge this in. Also, you gotta update the model for HexathonUser. Do you have plans to migrate the existing data in the database as well to support this.

**Comment by thaneesh-babu** [comment]
yeah like ayush said, registration should now interact with the hexathons service for team management stuff

File: services/hexathons/src/routes/team.ts [hunk 1]
```diff
@@ -154,7 +154,7 @@ teamRoutes.route("/user/:userId").get(
       return;
     }
 
-    const profiles = await ProfileModel.find({
+    const profiles = await HexathonUserModel.find({
       userId: {
         $in: team.members,
       },
```
Comment (thaneesh-babu) [inline services/hexathons/src/routes/team.ts:160]: change this term to `hexathonUsers` and the find call should match the hexathon as well, rn hexathon users from all hexathons are taken as valid profiles

**Review by thaneesh-babu — CHANGES_REQUESTED** [review]

**Pull request merged**